src/settings/dto/update-setting.dto.ts

```diff
@@ -0,0 +1,4 @@
+import { PartialType } from '@nestjs/mapped-types';
+import { CreateSettingDto } from './create-setting.dto';
+
+export class UpdateSettingDto extends PartialType(CreateSettingDto) {}
```

Comment:
If the code assumes that DataType should always be provided in the update API, then it shouldn't be optional at the DTO level